feat: add option to disable blob verification

fty4 opened the pull request:
I have the issue that the address `sigstore.dev` is not shared on my runner (we do not like to open it).
So i would like to disable the verification by option.

Therefore I created a input to disable the verification in the action.
With the default Action configuration the sensible verification is enabled for sure.

fty4 committed:
feat: add input verify_blob to add option to disable verification

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
fix: move curl outside verify_blob statement

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 commented:
If desired i can add a github wf test - just let me know

cpanato commented on action.yml:
```diff
@@ -5,6 +5,9 @@ branding:
 inputs:
+  verify_blob:
+    description: "determines whether the download blob should be verified (default: true)"
+    default: 'true'
```
if you disable that, i think you should also not install the cosign actions

fty4 replied:
You are absolutely right - I fixed that.

cpanato reviewed (changes requested):
thanks for that

add a test case for that, please

fty4 committed:
chore: disable cosign installation when verify blob disabled

As from cpanato suggested the installation of cosign should be skipped.

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
chore: add gh wf test test_ct_action_noverify

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 commented:
@cpanato  Added the requested changes

cpanato commented on action.yml:
```diff
@@ -5,6 +5,9 @@ branding:
 inputs:
+  verify_blob:
+    description: "determines whether the download blob should be verified (default: true)"
+    default: 'true'
```
add the `required: false`


can you also make sure that will work like `verify_blob: false` and `verify_blob: 'false'` ?

fty4 replied:
Tested both cases with [act](https://github.com/nektos/act) and received the [messages](https://github.com/helm/chart-testing-action/pull/131/commits/6acc8cc67bee951c83257172c69f7c6d7d503ee1) from the log as expected -> works with both (with and without quotes).

cpanato commented on .github/workflows/test-action.yml:
```diff
@@ -65,3 +65,34 @@ jobs:
+  test_ct_action_noverify:
+    runs-on: ubuntu-latest
+
+    name: Install chart-testing and test presence in path
```
change this name to something more meaningful for this test

fty4 replied:
Missed to update name after coping from basic test - updated the value.

fty4 committed:
Merge remote-tracking branch 'origin' into feat/verify-blob

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
chore: add required key for input verify_blob

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
fix: set noverify test name to more meaningful value

The name of the test_ct_action_noverify test did not receive any customizing for the name. No there is a unique name to describe what is tested.

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
chore: adding verify message

Adding messages to determinate if blob was verified.

Also moving Install message out of verify if statement

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
fix: update test version to v3.10.0

Updating version required after merging main to PR.

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 commented:
@cpanato All requests are now implemented.

cpanato commented:
can you rebase?

fty4 committed:
Merge remote-tracking branch 'origin/main' into feat/verify-blob

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 committed:
fix: using latest version v3.12.0 for no-verify test

Signed-off-by: Marco Lecheler <marco.lecheler@mercedes-benz.com>

fty4 commented:
@cpanato ✅

cpanato commented on .github/workflows/test-action.yml:
```diff
@@ -111,3 +111,36 @@ jobs:
+          exit 1
+        else
+          exit 0
+        fi
```
nit: new line

fty4 replied:
✅

fty4 commented on .github/workflows/test-action.yml:
```diff
@@ -111,3 +111,36 @@ jobs:
+          exit 1
+        else
+          exit 0
+        fi
```
```suggestion
        fi

```

fty4 committed:
fix: add new line

Signed-off-by: Marco Lecheler <marco@task.media>

fty4 commented:
Recreated and updated find new PR #205
